### PR TITLE
Add Foreman bats CI box

### PR DESCRIPTION
**Pull request by ehelms**
Start of box definition to perform Foreman bats CI the same as Katello. This goes along with some in progress work (https://github.com/theforeman/foreman-infra/pull/323/commits/54958a6d73c59e4b8b265e97b6208af48344a96e).

### CODE REVIEW

File: boxes.yaml.example [hunk 1]
```diff
@@ -47,6 +47,17 @@ centos7-bats-ci:
         - "fb-content-katello.bats"
         - "fb-destroy-organization.bats"
         - "fb-finish-katello.bats"
+
+centos7-foreman-bats-ci:
+  box: centos7-bats
+  ansible:
+    playbook: 'playbooks/bats_pipeline_foreman_nightly.yml'
```
Comment (ekohl) [inline boxes.yaml.example]: Should we rename ```playbooks/bats_pipeline_nightly.yml``` to ```playbooks/bats_pipeline_katello_nightly.yml```?

File: boxes.yaml.example [hunk 2]
```diff
@@ -47,6 +47,17 @@ centos7-bats-ci:
...
+    group: 'bats'
+    variables:
+      bats_tests:
+        - "fb-test-foreman.bats"
+        - "fb-finish-katello.bats"
```
Comment (ekohl) [inline boxes.yaml.example]: Should we rename this one to ```fb-finish-foreman.bats``` or just ```fb-finish.bats```?

File: boxes.yaml.example [hunk 3]
```diff
@@ -38,15 +38,26 @@ katello-client:
 centos7-bats-ci:
```
Comment (ekohl) [inline boxes.yaml.example]: It might be nicer to rename this to centos7-katello-bats-ci

**Comment by ekohl** [comment]
I just tried this and I got an error with the swapfile.

```
[root@centos7-foreman-bats-ci ~]# swapon /swapfile
swapon: /swapfile: swapon failed: Invalid argument
```

It refuses because there are holes in it and I think that's because I use btrfs on /var/lib/images

File: boxes.yaml.example [hunk 4]
```diff
@@ -38,15 +38,26 @@ katello-client:
...
   box: centos7-bats
```
Comment (ekohl) [inline boxes.yaml.example]: I don't think the centos7-bats box has much use anymore.

File: boxes.yaml.example [hunk 5]
```diff
@@ -38,15 +38,26 @@ katello-client:
...
   ansible:
-    playbook: 'playbooks/bats_pipeline_nightly.yml'
+    playbook: 'playbooks/bats_pipeline_katello_nightly.yml'
     group: 'bats'
     variables:
```
Comment (ekohl) [inline boxes.yaml.example]: Can we move these variables to the playbook?

File: boxes.yaml.example [hunk 6]
```diff
@@ -38,15 +38,26 @@ katello-client:
...
       bats_tests:
         - "fb-test-foreman.bats"
         - "fb-test-katello.bats"
         - "fb-content-katello.bats"
         - "fb-destroy-organization.bats"
-        - "fb-finish-katello.bats"
+        - "fb-finish.bats"
+
+centos7-foreman-bats-ci:
+  box: centos7-bats
```
Comment (ekohl) [inline boxes.yaml.example]: This can also just use the ```centos7``` box.

File: boxes.yaml.example [hunk 7]
```diff
@@ -38,15 +38,26 @@ katello-client:
...
+  ansible:
+    playbook: 'playbooks/bats_pipeline_foreman_nightly.yml'
+    group: 'bats'
+    variables:
```
Comment (ekohl) [inline boxes.yaml.example]: Can these be moved to the playbook as well?

File: boxes.yaml.example (hunk 3 above)
Comment (ehelms) [reply boxes.yaml.example]: Note this will require a CI change -- https://github.com/theforeman/foreman-infra/pull/333

**Comment by ehelms** [comment]
This has been updated, but note that -- https://github.com/theforeman/foreman-infra/pull/333 will need to go in at the same time.

**Pull request merged**